protx/decorators.py

```diff
@@ -3,33 +3,49 @@
 from diskcache import Cache
 import os
 import logging
-from flask import request
+from flask import request, redirect, url_for
 import requests
 
 logger = logging.getLogger(__name__)
 
 cache = Cache("database_cache")
 
+def is_setup_complete(function, redirect_onboarding, *args, **kwargs):
+    try:
+        # making request directly to core django. alternatively, we could
+        # make the request to the request.get_host() (but then a bit
+        # tricky if it is local development and accessing cep.dev)
+        r = requests.get("http://django:6000/api/workbench/", cookies=request.cookies)
+        r.raise_for_status()
+        json_response = r.json()
+        if json_response['response']['setupComplete']:
+            return function(*args, **kwargs)
```

Comment:
having the return here in the try/catch is problematic.  this was fixed in main. i see it come in on your merge but then those fixes got lost in the refactoring